File: 07-Vaccine/test/mysql/conf/backend.js
const express = require('express')
const mysql = require('mysql')
const bodyParser = require('body-parser')

const db = mysql.createConnection({
        host: 'localhost',
        user: 'root',
        password: '123',
        database: 'test'
})
db.connect(err => {
        if (err) throw err
        console.log('Backend connected to MySQL database.')
})

const app = express()
app.use(bodyParser.json())

app.post('/login', (req, res) => {
        const { username, password } = req.body
        const query = "SELECT * FROM users WHERE login = '" + username + "' AND password = '" + password + "'"
        db.query(query, (err, result) => {
                if (err) {
                        console.error(err)
                        res.json({ success: false, errorMessage: err.message })
                        return
                }
                if (result) res.json({ success: result.length > 0, infos: result })
        })
})
app.get('/login', (req, res) => {
        const { username, password } = req.query
        const query = "SELECT * FROM users WHERE login = '" + username + "' AND password = '" + password + "'"
        db.query(query, (err, result) => {
                if (err) {
                        console.error(err)
                        res.json({ success: false, errorMessage: err.message })
                        return
                }
                if (result) res.json({ success: result.length > 0, infos: result })
        })
})
app.listen(8282, () => { console.log('MySQL backend listen on port 8282.') })

File: 07-Vaccine/test/sqlite/conf/backend.py
from flask import Flask, jsonify, request
import sqlite3


app = Flask(__name__)

@app.route("/login", methods=["POST", "GET"])
def login():
    if request.method == "POST":
        data = request.json
        username = data["username"]
        password = data["password"]
    elif request.method == "GET":
        username = request.args.get("username")
        password = request.args.get("password")

    conn = sqlite3.connect("test.db")
    cursor = conn.cursor()
    try:
        query = "SELECT * FROM users WHERE login = '" + username + "' AND password = '" + password + "'"
        cursor.execute(query)
        users = cursor.fetchall()
        success = len(users) > 0
        return jsonify({"success": success, "infos": users})
    except Exception as e:
        return jsonify({"success": False, "message": str(e)})
    finally:
        conn.close()


if __name__ == "__main__":
    app.run(host="0.0.0.0", port=8484)